CardsCashCasino/Manager/TexasHoldEmManager.cs

```
@@ -810,199 +813,249 @@ private static void OnTimeoutEvent(object source, ElapsedEventArgs e)
+            Pots[0].DecrementPot(Pots[Pots.Count - 1].Total); //reseting active pot value to empty
+            Pots[Pots.Count - 1].EligiblePlayers = new List<int>(Pots[0].EligiblePlayers); //copys the list of players eligible to win the now side pot
+
+            //removing all players that are all in from eligibility from main pot and any pots created in the future
+            for (int player = 0; player < allInPlayers.Count; player++)
             {
-                return Pots.Select(pot => pot.Total).ToList();
+                Pots[0].RemoveEligiblePlayer(allInPlayers[player]);
             }
 
-            /// <summary>
-            /// Resets the pots list to an empty state.
-            /// </summary>
-            public void ResetPots()
+            //if there is a second player that has gone all in this round, recursively add pots until player that has called max bet has been reached
+            if (currentBet != playerBets.Min())
             {
-                Pots.Clear();
+                CreateSidePots(currentBet, playerBets);
+            }
+
+            //if all players but one are all-in 
+            else if (numBets == 1)
+            {
+                Pots.RemoveAt(0); //remove redundant pot creation
+                Pots[0].PotType = PotType.MAIN;
+            }
+
+            else
+            {
+                //add remainder of bets from players not all-in into new main pot
+                AddToPot(currentBet, playerBets);
             }
         }
-        #endregion Methods
+
+        /// <summary>
+        /// Pays out the winnings to the player for the individual pot. 
+        /// </summary>
+        /// <param name="winners">Number of players that have won the pot.</param>
+        public int DistributePot(int winners, int potNum)
+        {
+            if (winners == 0)
+            {
+                throw new ArgumentException("Number of winners cannot be zero.");
```

under no circumstance should we throw an exception like this. instead, on the call to DistributePot, just confirm winners > 0 before calling and if not, just skip it because its errored out.

Throwing exceptions like this is scary, and can break the game if its something simple.